src/output-json-pgsql.c

```diff
@@ -47,7 +47,7 @@
 #include "output-json-pgsql.h"
 #include "rust.h"
 
-#define PGSQL_LOG_PASSWORDS BIT_U32(1)
+#define PGSQL_LOG_PASSWORDS BIT_U32(0)
```

Comment:
I think we can get rid of this line as cbindgen now export constants

Reply:
Not managing to make it work, failed at finding similar code in suri that could guide me...

Reply:
Could it it that it's not working because it uses the macros on the rust side, when declaring the constants?

Reply:
(if I expand the value instead of using the macros, I manage to make the code compile. Otherwise, I didn't find a way to make it work...)

Reply:
cf `APP_LAYER_PARSER_EOF_TS` defined in rust, used in C

Reply:
cbindgen should put it in rust/gen/rust-bindings.h 
cf commit 87d9c44ec504f459578fdb551b16a1015b8239c7

It looks like it works if you do
```
-pub const PGSQL_LOG_PASSWORDS: u32 = BIT_U32!(1);
+pub const PGSQL_LOG_PASSWORDS: u32 = 2;
```

@jasonish do you know why cbindgen does not work with this macro ?
And there are other examples of constants defined twice. (like `FLOW_DIR_REVERSED`)
Should we have a CI check for this ?

Reply:
> It looks like it works if you do
> 
> ```
> -pub const PGSQL_LOG_PASSWORDS: u32 = BIT_U32!(1);
> +pub const PGSQL_LOG_PASSWORDS: u32 = 2;
> ```

Yeah, I noticed that, as well. That was the only way to make it work.

Reply:
> cbindgen should put it in rust/gen/rust-bindings.h cf commit [87d9c44](https://github.com/OISF/suricata/commit/87d9c44ec504f459578fdb551b16a1015b8239c7)
> 
> It looks like it works if you do
> 
> ```
> -pub const PGSQL_LOG_PASSWORDS: u32 = BIT_U32!(1);
> +pub const PGSQL_LOG_PASSWORDS: u32 = 2;
> ```
> 
> @jasonish do you know why cbindgen does not work with this macro ? 

Cbindgen is a rather simple parser that doesn't expand macros like this. There is experimental support for it, but you have to be using the nightly rust compiler, so its not an option for us.

> And there are other examples of constants defined twice. (like `FLOW_DIR_REVERSED`) Should we have a CI check for this ?

I'm not sure what CI would be checking for? Duplicate constants?

Reply:
> Cbindgen is a rather simple parser that doesn't expand macros like this. There is experimental support for it, but you have to be using the nightly rust compiler, so its not an option for us.

Do you mean Cbindgen launches an external processus rustc nightly ? Or That we have to compile cbindgen with nightly

> I'm not sure what CI would be checking for? Duplicate constants?

Indeed

Reply:
> Do you mean Cbindgen launches an external processus rustc nightly ? Or That we have to compile cbindgen with nightly

It uses rustc, so nightly has to be set as your default toolchain, or the default rustc in your path.

Reply:
So, what is best ? Should we avoid macros in const definitions ? So that they can be used in C ?

Reply:
The last is closest to the `BIT_U32` macro, but overall I see this as a step back :-/

Reply:
So what should be the preferable approach in this case, for now? Leave the duplicates, or keep it just Rust, with the one that is closest to the macro?

Reply:
I just created https://github.com/eqrion/cbindgen/issues/735

Reply:
I created https://redmine.openinfosecfoundation.org/issues/5012, not first priority